src/domain/community/virtual-contributor/virtual.contributor.resolver.fields.ts

```diff
@@ -101,21 +105,30 @@ export class VirtualContributorResolverFields {
     );
   }
 
-  @AuthorizationAgentPrivilege(AuthorizationPrivilege.READ_ABOUT)
   @ResolveField('knowledgeBase', () => IKnowledgeBase, {
     nullable: true,
     description: 'The KnowledgeBase being used by this virtual contributor',
   })
   @UseGuards(GraphqlGuard)
   async knowledgeBase(
-    @Parent() virtualContributor: VirtualContributor
+    @Parent() virtualContributor: VirtualContributor,
+    @CurrentUser() agentInfo: AgentInfo
   ): Promise<IKnowledgeBase> {
-    return this.virtualContributorService.getKnowledgeBaseOrFail(
-      virtualContributor
+    const knowledgeBase =
+      await this.virtualContributorService.getKnowledgeBaseOrFail(
+        virtualContributor
+      );
+
+    this.authorizationService.grantAccessOrFail(
+      agentInfo,
+      knowledgeBase.authorization,
+      AuthorizationPrivilege.READ_ABOUT,
```

Comment:
Wouldn't that mean, in the long run, that we need to have VirtualContributor --> About --> KnowledgeBase, with READ_ABOUT on VC, Then READ everywhere below?